kong/plugins/opentelemetry/handler.lua

```diff
@@ -42,6 +43,11 @@ end
 
 
 function OpenTelemetryHandler:log(conf)
+  -- Read resource attributes variable
+  if conf.resource_attributes then
+    conf["compiled_resource_attributes"] = otel_utils.read_resource_attributes(conf.resource_attributes)
```

Comment:
I would recommend against injecting values into conf that are not part of the configuration (and defined in its schema). My suggestion here is to use an additional table and pass it alongside `conf` to the otel_traces and otel_logs log functions, for example for traces it would look something like:

```
  local options = {
    compiled_resource_attributes = otel_utils.read_resource_attributes(conf.resource_attributes)
  }
```
```
otel_traces.log(conf, options)
```

The `Queue.enqueue` function in `traces.lua` accepts a single parameter for the handler (currently `conf`), so both `conf` and `options` will have to be packed in a table like: `local params = {conf = conf, options = options}` and passed there. These can then be accessed from `http_export_traces` where the compiled attributes can be used if available. It's a bit more complex but it allows to keep things more isolated, what do you think?